tsconfig.json

```diff
@@ -0,0 +1,3 @@
+{
+
```

Comment:
Would it be sensible to have some defaults here,  like:

```javascript
{
  "moduleResolution": "node"
}
```

It is hard to do such a general `tsconfig` file, but if left empty it will not contribute to any changes in configuration speed.

Reply:
I agree, I was not intending on adding anything at all really, because the root of the monorepo is not itself a ts project.
But for some reason VSCode was complaining about missing tsconfig.json - maybe via the linter.
I think a tsconfig file without specifying any source folders also is invalid. So as discussed I think its best to leave this blank for now.

Reply:
One possible trick would be to use `tsconfig.base.json` so TS and VSCode don't pick up on it but you can use it to extend your own tsconfig, but not sure it is worth it for our usecase

Reply:
Yes I toyed with a reusable package earlier but dropped it.

That error I saw was only in pioneer/ so I think the culprit is this line, which seems un-necessary:

https://github.com/Joystream/joystream/blob/5700ce86922e2bbd34cab68488d603b5c06b7068/pioneer/.eslintrc.js#L9

commenting out the `projects` array seems to resolve it, and pioneer still builds.

any ideas on this @Lezek123

Reply:
Removed it in 49daa93cc2c3e735aa5859835a999254c667741d

Reply:
Well that was a FAIL on my side!

```
Error: Error while loading rule '@typescript-eslint/await-thenable': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Occurred while linting /home/runner/work/joystream/joystream/pioneer/babel.config.js
```

IDE was happy, didn't check linter though.. will revert